src/ADSB/ADSBVehicleManager.cc

```diff
@@ -121,72 +119,115 @@ void ADSBTCPLink::_hardwareConnect()
-            emit adsbVehicleUpdate(adsbInfo);
-        } else if (values[1] == QStringLiteral("4")) {
-            bool icaoOk, headingOk;
-
-            uint32_t    icaoAddress =   values[4].toUInt(&icaoOk, 16);
-            double      heading =       values[13].toDouble(&headingOk);
-
-            if (!icaoOk || !headingOk) {
-                return;
-            }
-
-            ADSBVehicle::VehicleInfo_t adsbInfo;
-            adsbInfo.icaoAddress = icaoAddress;
-            adsbInfo.heading = heading;
-            adsbInfo.availableFlags = ADSBVehicle::HeadingAvailable;
-            emit adsbVehicleUpdate(adsbInfo);
-        } else if (values[1] == QStringLiteral("1")) {
-            bool icaoOk;
-
-            uint32_t icaoAddress = values[4].toUInt(&icaoOk, 16);
-            if (!icaoOk) {
-                return;
-            }
-
-            ADSBVehicle::VehicleInfo_t adsbInfo;
-            adsbInfo.icaoAddress = icaoAddress;
-            adsbInfo.callsign = values[10];
-            adsbInfo.availableFlags = ADSBVehicle::CallsignAvailable;
-            emit adsbVehicleUpdate(adsbInfo);
+        bool icaoOk;
+        int msgType = line.at(4).digitValue();
+        if (msgType == -1) {
+            qCDebug(ADSBVehicleManagerLog) << "ADSB Invalid message type " << line.at(4);
+            return;
+        }
+        // Skip unsupported mesg types to avoid parsing
+        if (msgType == 2 || msgType > 6) {
+            return;
+        }
+        qCDebug(ADSBVehicleManagerLog) << " ADSB SBS-1 " << line;
+        QStringList values = line.split(QChar(','));
+        uint32_t icaoAddress = values[4].toUInt(&icaoOk, 16);
+
+        if (!icaoOk) {
+            return;
+        }
+
+        ADSBVehicle::ADSBVehicleInfo_t adsbInfo;
+        adsbInfo.icaoAddress = icaoAddress;
+
+        switch (msgType) {
+        case 1:
+        case 5:
+        case 6:
+            _parseAndEmitCallsign(adsbInfo, values);
+            break;
+        case 3:
+            _parseAndEmitLocation(adsbInfo, values);
+            break;
+        case 4:
+            _parseAndEmitHeading(adsbInfo, values);
+            break;
         }
     }
 }
+
+void ADSBTCPLink::_parseAndEmitCallsign(ADSBVehicle::ADSBVehicleInfo_t &adsbInfo, QStringList values)
+{
+    QString callsign = values[10].trimmed();
+    if (callsign.isEmpty()) {
+        return;
+    }
+
+    adsbInfo.callsign = callsign;
+    adsbInfo.availableFlags = ADSBVehicle::CallsignAvailable;
+    emit adsbVehicleUpdate(adsbInfo);
+}
+
+void ADSBTCPLink::_parseAndEmitLocation(ADSBVehicle::ADSBVehicleInfo_t &adsbInfo, QStringList values)
+{
+    bool altOk, latOk, lonOk;
+    int modeCAltitude;
+
+    QString altitudeStr = values[11];
+    // Altitude is either Barometric - based on pressure, in ft
+    // or HAE - as reported by GPS - based on WGS84 Ellipsoid, in ft
+    // If altitude ends with H, we have HAE
+    if (altitudeStr.endsWith('H')) {
```

Comment:
Extra return duplicates the error checking further down the code. Also better to not duplicate string conversion code. I would change as follows:
```
    QString altitudeStr = values[11];
    // Altitude is either Barometric - based on pressure, in ft
    // or HAE - as reported by GPS - based on WGS84 Ellipsoid, in ft
    // If altitude ends with H, we have HAE
    // There's a slight difference between Barometric alt and HAE, but it would require
    // knowledge about Geoid shape in particular Lat, Lon. It's not worth complicating the code
    if (altitudeStr.endsWith('H')) {
        altitudeStr.chop(1);
    }
    modeCAltitude = altitudeStr.toInt(&altOk);
```

Reply:
I wanted to bail out as soon as possible. But, sure, it reduces a few CPU ticks but shadows code readability. Clean code FTW.